### PR TITLE
1:10 doesnt lint, optionally

### LINKED ISSUE #1155: implicit_integer_linter shouldn't lint 1:10 (optionally?)
```
lintr::lint("1:10\n", lintr::implicit_integer_linter())
# <text>:1:2: style: Integers should not be implicit. Use the form 1L for integers or 1.0 for doubles.
# 1:10
# ~^
# <text>:1:5: style: Integers should not be implicit. Use the form 1L for integers or 1.0 for doubles.
# 1:10
#   ~~^
```

but `typeof(1:10)` is `"integer"`, and the performance is essentially identical:

```
microbenchmark::microbenchmark(
  1L:10L,
  1L:10,
  1:10L,
  1:10,
  times = 1e6
)

# Unit: nanoseconds
#    expr min  lq     mean median  uq      max neval
#  1L:10L 199 242 479.8983    257 284 89303201 1e+06
#   1L:10 197 237 376.0418    251 278 10444161 1e+06
#   1:10L 198 239 354.6013    252 280 10678523 1e+06
#    1:10 190 233 405.9966    246 274 14666175 1e+06
```

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #1155

### CODE REVIEW

File: R/implicit_integer_linter.R [hunk 1]
```diff
@@ -30,18 +37,28 @@
...
+#' lint(
+#'   text = "1:10",
+#'   linters = implicit_integer_linter(allow_colon = TRUE)
+#' )
+#'
 #' @evalRd rd_tags("implicit_integer_linter")
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
-implicit_integer_linter <- function() {
+implicit_integer_linter <- function(allow_colon = FALSE) {
```
Comment (IndrajeetPatil) [inline R/implicit_integer_linter.R:48]: Shouldn't the default argument be `TRUE` for this parameter? 
Which cases that should produce a lint will be missed out on, if we default to `TRUE`?

Comment (MichaelChirico) [reply R/implicit_integer_linter.R:48]: Done for back-compatiblity... current HEAD implicitly uses `FALSE`.

I'd be open to changing it later, but at least for this release we should use this default.

Comment (IndrajeetPatil) [reply R/implicit_integer_linter.R:48]: SGTM.

Can you please file a new issue to remember to do this? And maybe also choose the appropriate milestone for the issue?

**Review by IndrajeetPatil — APPROVED** [review]

File: R/implicit_integer_linter.R (hunk 1 above)
Comment (MichaelChirico) [reply R/implicit_integer_linter.R:48]: #1694. Let's hold off on choosing a milestone, but maybe 3.1.0? Let's see if we get any feedback after 3.0.2 (I guess it's really 3.0.3 now) is released with this PR.

**Pull request merged**